src/main/java/org/opentest4j/MultipleFailuresError.java

```diff
@@ -34,15 +36,15 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final List<AssertionError> failures = new ArrayList<AssertionError>();
+	private final List<Throwable> failures = new ArrayList<Throwable>();
 
 	private final String heading;
 
 	public MultipleFailuresError(String heading) {
 		this.heading = isBlank(heading) ? "Multiple Failures" : heading.trim();
 	}
 
-	public void addFailure(AssertionError failure) {
+	public void addFailure(Throwable failure) {
```

Comment:
@kcooney Good catch: Exceptions should be immutable! Therefore, we should indeed pass in the list of failures into the constructor, copy and store it.

@gaganis Can you do this change in this PR? Alternatively, we could create a separate issue to address it.